int32 left_distance
int32 right_distance